iron-list.html

```diff
@@ -486,19 +490,85 @@
         }
       }
 
-     // update the position of the items
+      // update the position of the items
       this._positionItems();
 
       // set the scroller size
       this._updateScrollerSize();
+
+      if (this._increasePoolIfNeeded()) {
+        // set models to the new items
+        this.async(this._update);
+      }
+    },
+
+    /**
+     * Creates a pool of DOM elements and attaches them to the local dom.
+     */
+    _createPool: function(size) {
+      var physicalItems = new Array(size);
+
+      this._ensureTemplatized();
+
+      for (var i = 0; i < size; i++) {
+        var inst = this.stamp(null);
+
+        // First element child is item; Safari doesn't support children[0]
+        // on a doc fragment
+        physicalItems[i] = inst.root.querySelector('*');
+        Polymer.dom(this).appendChild(inst.root);
+      }
+
+      return physicalItems;
     },
 
+    /**
+     * Increases the pool size. That is, the physical items in the DOM.
+     * This function will allocate twice as many physical items
+     * (limited by `MAX_PHYSICAL_COUNT`) if the content size is not at least
+     * three times the size of the viewport.
+     */
+    _increasePoolIfNeeded: function() {
+      if (this._physicalSize > this._viewportSize * 3) {
+        return false;
+      }
+
+      var nextPhysicalCount = Math.min(
+          // double the size of the physical items
+          this._physicalCount * 2,
+          this._virtualCount,
+          MAX_PHYSICAL_COUNT
+        );

```

Comment:
This is a valid point. I agree that if we know the target physical size, then we could use the `physicalAverage` to estimate how many items we should add to the pool. I will still keep `MAX_PHYSICAL_COUNT` since we might hit a worst case scenario if the height of the new physical items is zero.